compose.yaml

```diff
@@ -75,46 +81,39 @@ services:
   localweb:
     profiles:
       - offline
-    extends:
-      file: common-app.yaml
-      service: rails
+    image: terrastories-offline-test
```

Comment:
This didn't work for me:

```
 ✘ tileserver Error                                                      0.4s 
 ✘ offline Error                                                         0.4s 
 ✘ localweb Error                                                        0.4s 
Error response from daemon: pull access denied for terrastories-offline-test, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
```

I think we still need to provide the build context for this method of building terrastories with tileserver-gl, no?

Reply:
I get some copy errors trying to build `terrastories-offline-test`:

```sh
 => ERROR [prod 4/9] COPY ./rails /api                                                                                                                              0.0s
 => ERROR [builder 2/6] COPY ./rails /api                                                                                                                           0.0s
 => ERROR [builder 4/6] COPY ./map /api/public/map                                                                                                                  0.0s
------
 > [prod 4/9] COPY ./rails /api:
------
------
 > [builder 2/6] COPY ./rails /api:
------
------
 > [builder 4/6] COPY ./map /api/public/map:
------
Dockerfile:54
--------------------
  52 |     
  53 |     # Configure Map Package
  54 | >>> COPY ./map /api/public/map
  55 |     RUN curl -fsSL "https://t.ly/OTZpR" -o "/api/public/map/terrastories-map/tiles.pmtiles"
  56 |     RUN curl -fsSL "https://github.com/protomaps/basemaps-assets/archive/refs/heads/main.tar.gz" | tar xzf - -C /api/public/map/terrastories-map --strip-components=1 basemaps-assets-main/fonts basemaps-assets-main/sprites
--------------------
ERROR: failed to solve: failed to compute cache key: failed to calculate checksum of ref 78505b1b-92e8-467e-a316-99d544772b2e::1xp0b3v6p4ws3u1obo68j56w6: "/map": not found
```

Aside from the weird ./rails -> ./api copy error, it seems we probably shouldn't need to copy anything relating to a default offline pmtiles for the terrastories-with-tileserver flavor.

Reply:
> `docker build -t terrastories-offline-test rails/.`

This is not quite right anymore.

`docker build -t terrastories-offline-test -f rails/Dockerfile .` The context needs to be from the root folder, not nested within `/rails`.

> it seems we probably shouldn't need to copy anything relating to a default offline pmtiles for the terrastories-with-tileserver flavor.

This is not quite right. The intention is to use the same image for tileserver, protomaps, or mapbox. We always bundle offline-ready pmtiles for the image so it JustWorks™ if you're trying to set it up for the first time on your own with no other map configuration.

Reply:
That mostly makes sense - it only partially does not insofar as I've only gotten to this step by having tileserver map configuration files to begin with (hence bypassing the abort-if-tileserver-configuration-isn't-found check in the `bin/install` script). 

That said, I'm fine with always bundling a pmtiles file.

Reply:
Anyway, having now successfully built `terrastories-offline-test`, and seeing the expected console messages from `bin/serve`,

```sh
[+] Running 6/6
 ✔ Volume "terrastories_postgres_data"  Created                                                                                   0.0s 
 ✔ Volume "terrastories_bundler"        Created                                                                                   0.0s 
 ✔ Container terrastories-tileserver-1  Started                                                                                   0.1s 
 ✔ Container terrastories-db-1          Started                                                                                   0.1s 
 ✔ Container terrastories-localweb-1    Started                                                                                   0.0s 
 ✔ Container terrastories-offline-1     Started                                                                                   0.0s 
You can access Terrastories at: http://rudodev
You can access Tileserver at: http://rudodev/tileserver
```

Running either of these URLS in the browser (or localhost, 127.0.0.1, terrastories.local) yields an Nginx 404 Not Found page.

Reply:
Ah, I think I understand the confusion. We will no longer be building a docker image using any of the bin/ scripts. This is a huge change from our current standard. Instead, the bin scripts will pull down prebuilt images from Docker hub.

`terrastories-offline-test` is a stopgap image that needs to be locally built in order to test using the setup and install scripts. Normally, it would read `terrastories/terrastories:latest`.

Reply:
And I might be blanking on how to properly test this branch currently! I have been using the `bin/` scripts..

Reply:
Can you check the contents of `/etc/hosts` and `.env.offline` and post them here?

Reply:
`.env.offline` has the following var `HOST_HOSTNAME=rudodev` and my `/etc/hosts` contains the same for 127.0.0.1. I just removed any other localhost hostnames that were defined (like `terrastories.local`) to see if it would make any difference, but the result is the same on my end.

Reply:
What happens if you try rudodev:3000?